### PR TITLE
Moved method definition to interface

### PR DESCRIPTION
**Pull request by core23**
### Changelog

``` markdown
### Changed
- Moved public methods of `User` to `UserInterface`
```
### Subject

Moved all public methods of the `User` class to `UserInterface`
### To do

- [X] Moved methods to interface

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 CHANGELOG
 =========
 
+### 2.x to 3.x
```
Comment (Soullivaneuh) [inline CHANGELOG.md]: The changelog does not follow this format anymore.

**Comment by Soullivaneuh** [comment]
@core23 I'm not sure to merge this one right now because I would like to split the user model to make some fields optional, like social ones for example.

**Comment by OskarStark** [comment]
Stuck Style CI build @Soullivaneuh

**Comment by core23** [comment]
Think we should wait for the FOSUserBundle release before merging this.

**Comment by core23** [comment]
Should be merged after https://github.com/sonata-project/SonataUserBundle/pull/869

**Pull request merged**